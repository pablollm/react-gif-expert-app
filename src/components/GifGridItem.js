import React from 'react'

export const GifGridItem = ({gif}) => {
  return (
        <div className="card">
            <img src={gif.url} alt={gif.title}/>
            <p className='card-title'>{gif.title}</p>
        </div>
  )
}
