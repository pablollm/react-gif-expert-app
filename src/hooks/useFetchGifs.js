import { useState,useEffect } from "react";
import { getGifs } from "../services/getGifs";

import React from 'react'

export const useFetchGifs = (category) => {

    const [state, setState] = useState({   
        data: [],
        loading: true
     });
      useEffect(() => {
        getGifs(category)
        .then(
            (gifs) => {
                setState({
                    data: gifs,
                    loading: false
                });
            })
    }, [category])

  return state
}
