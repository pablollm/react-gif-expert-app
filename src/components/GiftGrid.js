import React from 'react'
import { useEffect ,useState} from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../services/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GiftGrid = ({category}) => {
const {data:gifs,loading} = useFetchGifs(category);
    
   
  return (
    <>
    <h3>{category}</h3>
    {loading && <p>Loading...</p>}
     <div className='card-grid'>
      {
        gifs.map(gif => (  <GifGridItem key={gif.id} gif={gif}/>))
     }
     </div>

    </>
  )
}

GiftGrid.propTypes = {
category: PropTypes.string.isRequired
}