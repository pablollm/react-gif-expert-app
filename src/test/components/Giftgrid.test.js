import React from 'react';
import {shallow} from 'enzyme';
import { GiftGrid } from '../../components/GiftGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GiftGrid/>', () => {
    const category = 'One Punch';

    test('debe de mostrar el componente correctamente', () => {
            useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GiftGrid  category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imagenes en useFetchGifs', () => {
        const gifs = [{ id: 'ABC' , url: 'https://media1.giphy.com/media/l0HlJn1y7z2QkQ7p2/200w.gif',title: 'One Punch'}];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
         const wrapper = shallow(<GiftGrid  category={category}/>);
         expect(wrapper).toMatchSnapshot();
         expect(wrapper.find('p').exists()).toBe(false);
         expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})