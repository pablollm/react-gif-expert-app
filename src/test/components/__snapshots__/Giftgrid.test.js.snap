// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGrid/> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GiftGrid/> debe de mostrar items cuando se cargan imagenes en useFetchGifs 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      gif={
        Object {
          "id": "ABC",
          "title": "One Punch",
          "url": "https://media1.giphy.com/media/l0HlJn1y7z2QkQ7p2/200w.gif",
        }
      }
      key="ABC"
    />
  </div>
</Fragment>
`;
